libraries/chain/include/eosio/chain/producer_schedule.hpp

```diff
@@ -33,6 +33,7 @@ namespace eosio { namespace chain {
 
       shared_producer_schedule_type& operator=( const producer_schedule_type& a ) {
          version = a.version;
+         producers.clear();
          producers.reserve( a.producers.size() );
          for( const auto& p : a.producers )
             producers.push_back(p);
```

Comment:
Github does not appear to let me add a comment outside of changed code...

Line 59, I think should have `if( a.producers.size() != b.producers.size()) return false;`

Reply:
agreed